### PR TITLE
TransactionChain pagination in the explorer

### LINKED ISSUE #764: Add pagination in the explorer's chain page
### Is your feature request related to a problem?

A transaction chain might contain thousands of transactions. We do not want to load them all on page load. 

### Describe the solution you'd like

Implement an infinite scrolling (because the existing pagination function does not take a limit and an offset but a paging_address)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bchamagne**
# Description

A transaction chain might contain thousands of transactions. We do not want to load them all on page load.
This PR add an infinite scrolling to this page. 
I removed the form option "find last" because it is now the default behaviour. There's no reason to see only a subset of a transaction chain.

Fixes #764

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tests added to the suite 
Tests on the explorer for every branch of code 

Screenshot 2022-12-13 at 16-10-16 ARCHETHIC
Screenshot 2022-12-13 at 16-10-49 ARCHETHIC
Screenshot 2022-12-13 at 16-11-03 ARCHETHIC
Screenshot 2022-12-13 at 16-11-15 ARCHETHIC
Screenshot 2022-12-13 at 16-11-48 ARCHETHIC

# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/archethic/db/embedded_impl.ex [hunk 1]
```diff
@@ -200,6 +200,19 @@ defmodule Archethic.DB.EmbeddedImpl do
...
+  @doc """
+  Get a transaction chain - anti-chronological order
+
+  The returned values will be splitted according to the pagination state presents in the options
+  """
+  @spec get_transaction_chain_desc(address :: binary(), fields :: list(), opts :: list()) ::
+          {transactions_by_page :: list(Transaction.t()), more? :: boolean(),
+           paging_state :: nil | binary()}
+  def get_transaction_chain_desc(address, fields \\ [], opts \\ [])
+      when is_binary(address) and is_list(fields) and is_list(opts) do
```
Comment (netboz) [inline lib/archethic/db/embedded_impl.ex:212]: In erlang, when chaining logical test with `and`, it used to be better to use `and_also` as the tests evaluation will then stop as soon as one of the subtest fails.
I think it is still the case : https://elixirforum.com/t/why-not-and-andalso-and-or-orelse/22156

File: lib/archethic/db/embedded_impl/chain_reader.ex [hunk 2]
```diff
@@ -239,6 +269,51 @@ defmodule Archethic.DB.EmbeddedImpl.ChainReader do
     {transactions, more?, paging_state}
   end
 
+  # in order to read the file sequentially in DESC (faster than random access)
+  # we have to determined the correct paging_state and limit_address
```
Comment (netboz) [inline lib/archethic/db/embedded_impl/chain_reader.ex:273]: typo `determined` :)

File: lib/archethic/db/embedded_impl.ex (hunk 1 above)
Comment (bchamagne) [reply lib/archethic/db/embedded_impl.ex:212]: There is no andalso in elixir

**Review by netboz — APPROVED** [review]

File: lib/archethic/p2p/message.ex [hunk 3]
```diff
@@ -1360,13 +1360,21 @@ defmodule Archethic.P2P.Message do
...
         },
         _
       ) do
     {chain, more?, paging_state} =
-      tx_address
-      |> TransactionChain.get([], paging_state: paging_state)
+      case order do
+        :asc ->
+          tx_address
+          |> TransactionChain.get([], paging_state: paging_state)
```
Comment (samuelmanzanera) [inline lib/archethic/p2p/message.ex:1372]: Why not using the option list to specify the order as well ;)

File: lib/archethic_web/live/chains/transaction_live.ex [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    end
+  end
+
+  defp get_paginated_transaction_chain(encoded_address) do
+    with {:ok, addr} <- Base.decode16(encoded_address, case: :mixed),
+         true <- Crypto.valid_address?(addr),
+         {:ok, %Transaction{address: last_address}} <- Archethic.get_last_transaction(addr),
+         {:ok, chain_length} <- Archethic.get_transaction_chain_length(last_address),
+         {:ok, chain} <- paginate_chain(last_address, nil),
+         {:ok, %{uco: uco_balance}} <- Archethic.get_balance(last_address),
```
Comment (samuelmanzanera) [inline lib/archethic_web/live/chains/transaction_live.ex:82]: Nice to fix this one :+1: (for the balance)

**Comment by Neylix** [comment]
I have some error on the display, Is looks like it's displaying the last transaction but in reverse order before displaying the expected output
image

**Comment by bchamagne** [comment]
> I have some error on the display, Is looks like it's displaying the last transaction but in reverse order before displaying the expected output image

Nice catch! It happened because I did not serialize the order flag. It's fixed now

**Review by Neylix — APPROVED** [review]

**Review by samuelmanzanera — APPROVED** [review]

**Pull request merged**